SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPossession/CrossServerPossessionTest.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "CrossServerPossessionTest.h"
+
+#include "Containers/Array.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "GameFramework/PlayerController.h"
+#include "SpatialFunctionalTestFlowController.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/TestActors/TestMovementCharacter.h"
+#include "TestPossessionPawn.h"
+#include "TestPossessionPlayerController.h"
+
+/**
+ * This test tests 1 Controller possess over 1 Pawn.
+ *
+ * This test expects a load balancing grid and ACrossServerPossessionGameMode
+ * Recommand to use 2*2 load balancing grid because the position is written in the code
+ * The client workers begin with a player controller and their default pawns, which they initially possess.
+ * The flow is as follows:
+ *	Recommend to use PossessionGym.umap in UnrealGDKTestGyms project which ready for tests.
+ *  - Setup:
+ *    - Specify `GameMode Override` as ACrossServerPossessionGameMode
+ *    - Specify `Multi Worker Settings Class` as Zoning 2x2(e.g. BP_Possession_Settings_Zoning2_2 of UnrealGDKTestGyms)
+ *	  - Set `Num Required Clients` as 1
+ *  - Test:
+ *	  - Create a Pawn in first quadrant
+ *	  - Create Controller in other quadrant, the position is determined by ACrossServerPossessionGameMode
+ *	  - Wait for Pawn in right worker.
+ *	  -	The Controller possess the Pawn in server-side
+ *	- Result Check:
+ *    - ATestPossessionPlayerController::OnPossess should be called >= 1 times
+ *	  - the reason why accept > 1 is even we set `Num Required Clients` as 1, but still have 3 instance of clients in this case
```

Comment:
I don't understand this sentence

Reply:
There is a potential risk in this code: UnrealGDK\SpatialGDK\Source\SpatialGDKEditor\Private\SpatialGDKEditorModule.cpp line:329
`for (; SpatialTestIt; ++SpatialTestIt)
{
	NumberOfClients = FMath::Max(SpatialTestIt->GetNumRequiredClients(), NumberOfClients);
}
`
It set the `PIESettingsOverride.NumberOfClients = NumberOfClients;` as the bigest one of the test case `Num Required Clients`.
Here we have CrossServerMultiPossessionTest as the 3 clients, so whatever we set the CrossServerPossessionTest as 1 or 2 clients, it has 3 clients actually.
This is the reason we check ATestPossessionPlayerController::OnPossess should be called >= 1 times.